Sources/BraveWallet/Crypto/Transaction Confirmations/SwapTransactionConfirmationView.swift

```diff
@@ -0,0 +1,299 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import BraveCore
+import BigNumber
+import Strings
+import DesignSystem
+
+struct SwapTransactionConfirmationView: View {
+  
+  let parsedTransaction: ParsedTransaction?
+  let network: BraveWallet.NetworkInfo?
+  
+  let editGasFeeTapped: () -> Void
+  let advancedSettingsTapped: () -> Void
+  
+  @Environment(\.sizeCategory) private var sizeCategory
+  @Environment(\.colorScheme) private var colorScheme
+  @Environment(\.pixelLength) private var pixelLength
+  @ScaledMetric private var faviconSize = 48
+  private let maxFaviconSize: CGFloat = 72
+  @ScaledMetric private var assetIconSize: CGFloat = 40
+  private let maxAssetIconSize: CGFloat = 50
+  @ScaledMetric private var assetNetworkIconSize: CGFloat = 15
+  private let maxAssetNetworkIconSize: CGFloat = 20
+  
+  var body: some View {
+    VStack(spacing: 20) {
+      originAndFavicon
+      
+      tokenValueComparison
+
+      tokensView
+
+      networkFeeView
+    }
+    .padding(.bottom, 20)
+  }
+  
+  private var originAndFavicon: some View {
+    VStack {
+      if let originInfo = parsedTransaction?.transaction.originInfo {
+        Group {
+          if originInfo.origin == WalletConstants.braveWalletOrigin {
+            Image(uiImage: UIImage(sharedNamed: "brave.logo")!)
+              .resizable()
+              .aspectRatio(contentMode: .fit)
+              .foregroundColor(Color(.braveOrange))
+          } else {
+            originInfo.origin.url.map { url in
+              FaviconReader(url: url) { image in
+                if let image = image {
+                  Image(uiImage: image)
+                    .resizable()
+                    .scaledToFit()
+                } else {
+                  Circle()
+                    .stroke(Color(.braveSeparator), lineWidth: pixelLength)
+                }
+              }
+              .background(Color(.braveDisabled))
+              .clipShape(RoundedRectangle(cornerRadius: 4))
+            }
+          }
+        }
+        .frame(width: min(faviconSize, maxFaviconSize), height: min(faviconSize, maxFaviconSize))
+        Text(urlOrigin: originInfo.origin)
+          .foregroundColor(Color(.braveLabel))
+          .font(.subheadline)
+          .multilineTextAlignment(.center)
+          .padding(.top, 8)
+      }
+    }
+  }
+  
+  @ViewBuilder private var tokenValueComparison: some View {
+    if let minBuyAmount = Double(parsedTransaction?.ethSwap?.minBuyAmount ?? ""),
+       let sellAmount = Double( parsedTransaction?.ethSwap?.fromAmount ?? "") {
```

Comment:
let's add a check to make sure sellAmount is > 0 before we divide it?

Reply:
Fixed in [1c4138a](https://github.com/brave/brave-ios/pull/6826/commits/1c4138a1cc888815116a3bcfece0711369eb5233)